### ISSUE TITLE
Please improve the UX

### ISSUE THREAD

**Issue by SmurgBurglar** [sonic2kk/steamtinkerlaunch#849]
I installed this tool to use reshade but its so unclear that i just gave up.

**Comment by sonic2kk** [comment]
I'm not sure what you expect to happen from such a low-effort issue.

You should refer to the documentation for how to install reshade, and if you want a better UI feel free to make a pull request. Should you run into any issues, make a PR upstream at Yad to add/improve the features you want.

You should also be able to configure shaders from the commandline.

**Comment by SmurgBurglar** [comment]
Let me rephrase what i said as i came off as dismissive and rude, As a casual Linux user who tries to avoid the command line as much as possible Steam Tinker Launcher as a tool for "easy graphical configuration of game tools" seems perfect but the fact that you need to look through documentation where it can be difficult and confusing to find what your looking for defeats the purpose.

Take the Main Menu for example it has 48 buttons such as Shader repos, Game Shaders, Gamescope, Vortex Etc. it would be clearer if these options were contained in there own sections for Mods Tools, Shaders, VR, and if there were also a check box for advanced options on the bottom of the window that would hide  options most users wouldn't need, it would reduce clutter and streamline the interface.

**Comment by sonic2kk** [comment]
> As a casual Linux user who tries to avoid the command line as much as possible

SteamTinkerLaunch is not for casual users, it is for experienced users. It has a menu but also a powerful command line usage. If you want to avoid the commandline, I don't recommend SteamTinkerLaunch.

I'm sorry if you got the idea that SteamTinkerLaunch is for non-technical users, but they are not the target audience for this tool. They can't be, SteamTinkerLaunch is for tinkering and tinkering shouldn't be done by casual users. Actually the audience I would prefer to use STL are the ones who can also contribute back.

You should also keep in mind that SteamTinkerLaunch is a Bash script, it is *all* commandline at its core. The menu options call additional parts of the script which can broadly be accessed manually from the commandline as well. For example, the `steamtinkerlaunch settings` command, which shows the main menu, calls the same function that the Wait Requester button does. Likewise the buttons on the main menu, take One-Time Run: pressing this button *will actually run the `steamtinkerlaunch otr` command* (https://github.com/sonic2kk/steamtinkerlaunch/blob/master/steamtinkerlaunch#L5888), the same way a user can do from the terminal with `steamtinkerlaunch otr <appid>`.  

> the fact that you need to look through documentation where it can be difficult and confusing to find what your looking for defeats the purpose.

I don't understand, you're supposed to look through documentation for any software you're trying to use if available. There's a saying called "RTFM", which while a bit vulgar, applies broadly ("Read the F-ing Manual"). Always refer to the documentation to learn how to use software when available so you can get the most out of it.

The idea that you shouldn't have to read documentation doesn't make sense to me, I'm sorry. The purpose of STL is provide automated steps for users who already have a good understanding of the steps STL automates. It isn't a tool for users who are not experienced with tinkering, hence the name "SteamTinkerLaunch".

> Take the Main Menu for example it has 48 buttons such as Shader repos, Game Shaders, Gamescope, Vortex Etc. it would be clearer if these options were contained in there own sections for Mods Tools, Shaders, VR, and if there were also a check box for advanced options on the bottom of the window that would hide options most users wouldn't need, it would reduce clutter and streamline the interface.

This would be extremely difficult to do with Yad. It doesn't support tabs on Wayland for example, and conditional menu logic is very difficult to implement on a large scale with it. As I said before, it would be better to contribute to Yad upstream rather than complaining here. Or better yet if you think you can improve the UI with the current Yad limitations, please by all means do, I would happily review those kinds of PRs.

> Let me rephrase what i said as i came off as dismissive and rude, As a casual Linux user who tries to avoid the command line as much as possible

It was, but I think the main issue on top of the phrasing is that perhaps you made this issue without doing some research on Yad and the technical limitations of what can be done with it. The fact that there is *any* conditional logic at all for parts of the menu is hacked together and I don't even fully understand how Frostworx did it way back before I took over. It is not something I think is a good idea to extend for a checkbox as it would be very prone to breaking when applied on a large scale.

Yad isn't the best choice for a UI like this but I don't think there's a solid alternative for Bash (yet). Though instead of an alternative, improving Yad upstream to allow any Bash developer to make better, more traditional interfaces would be nice, within reason of course. Bash is the best language choice for STL imo so I am **not** interested in switching (very compatible with all systems and the Steam Linux Runtime limitations, which Valve's Proton script goes to great lengths to work around and does a lot of env var and commandline stuff itself, Bash also means it's much more straightforward to allow commandline usage, which is a very important feature of STL, and finally Bash is probably the most accessible language for tinkerers and they'd probably have their own Bash scripts laying around too).
